bokeh/_json_encoder.py

```diff
@@ -79,4 +79,4 @@ def default(self, obj):
             return self.transform_python_types(obj)
 
 def serialize_json(obj, encoder=BokehJSONEncoder, **kwargs):
-    return json.dumps(obj, cls=encoder, allow_nan=False, **kwargs)
+    return json.dumps(obj, cls=encoder, allow_nan=True, **kwargs)
```

Comment:
As mentioned before by @bryevdv, we could try encoding the data using [UBSON](http://ubjson.org/). (note: I have never used this, but the [Why](http://ubjson.org/#why) section has me intrigued.)